neqo-crypto/src/secrets.rs

```diff
@@ -25,8 +25,8 @@ experimental_api!(SSL_SecretCallback(
 ));
 
 #[derive(Clone, Copy, Debug, FromRepr)]
-#[cfg_attr(windows, repr(i32))] // Windows has to be different, of course.
-#[cfg_attr(not(windows), repr(u32))]
+#[cfg_attr(all(windows, not(target_env = "gnu")), repr(i32))] // Use i32 for Windows MSVC
+#[cfg_attr(not(all(windows, not(target_env = "gnu"))), repr(u32))] // Use u32 for Windows MinGW (see https://bugzilla.mozilla.org/show_bug.cgi?id=1960482) and non-Windows
```

Comment:
Could also be written as:

```suggestion
#[cfg_attr(any(not(windows), target_env = "gnu"), repr(u32))] // Use u32 for MinGW (see https://bugzilla.mozilla.org/show_bug.cgi?id=1960482) and non-Windows
```

Reply:
I think that I prefer the straight negation you have already.  It's less readable on its own, but next to the first line, it's easier to reason about.